File: 228-summary-ranges/summary-ranges.go
import "strconv"

func summaryRanges(nums []int) []string {
    ranges := []string{}

    i := 0
    for i < len(nums) {
        start := nums[i]
        j := i
        for j+1 < len(nums) && nums[j+1] == nums[j]+1 {
            j++
        }

        end := nums[j]

        if start == end {
            ranges = append(ranges, strconv.Itoa(start))
        } else {
            ranges = append(ranges, strconv.Itoa(start)+"->"+strconv.Itoa(end))
        }
        i = j+1
    }

    return ranges
}

File: 228-summary-ranges/summary-ranges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        ranges = []
        if not nums:
            return ranges
        
        start = nums[0]
        
        for i in range(len(nums)):
            if i + 1 == len(nums) or nums[i+1] != nums[i] + 1:
                if nums[i] == start:
                    ranges.append(str(start))
                else:
                    ranges.append(f"{start}->{nums[i]}")
                
                if i + 1 < len(nums):
                    start = nums[i+1]

        return ranges